diff --git a/build/rpm/make_webpack.sh b/build/rpm/make_webpack.sh
new file mode 100755
index 0000000..ebdebce
--- /dev/null
+++ b/build/rpm/make_webpack.sh
@@ -0,0 +1,39 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 Red Hat.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# for more details.
+#
+set -e
+
+: ${1?"Usage: $0 version"}
+
+VER=$1
+shift
+PATCHES=$(echo $* | xargs --no-run-if-empty realpath)
+TMP=$(mktemp -d)
+
+pushd "$TMP"
+git clone --branch v$VER https://github.com/Netflix/vector vector-$VER
+cd vector-$VER
+
+for patch in $PATCHES
+do
+    echo Applying patch $patch
+    patch -p1 < $patch
+done
+
+npm install
+node_modules/webpack/bin/webpack.js --display-error-details --config webpack.prod.js
+popd
+
+tar czf vector_webpack-$VER.tar.gz -C "$TMP" vector-$VER/dist
+tar czf vector_testlibs-$VER.tar.gz -C "$TMP" vector-$VER/node_modules
diff --git a/build/rpm/vector.spec b/build/rpm/vector.spec
new file mode 100644
index 0000000..a4c899e
--- /dev/null
+++ b/build/rpm/vector.spec
@@ -0,0 +1,79 @@
+%global vector_version 2.0.0-beta.1
+
+Name:           vector
+Version:        2.0.0
+Release:        0.1.beta.1
+Summary:        On-host performance monitoring framework
+
+License:        ASL 2.0
+URL:            https://getvector.io
+Source0:        https://github.com/Netflix/vector/archive/v%{vector_version}/vector-%{vector_version}.tar.gz
+Source1:        vector_webpack-%{vector_version}.tar.gz
+Source2:        vector_testlibs-%{vector_version}.tar.gz
+Source3:        vector-httpd-conf
+Source4:        vector-nginx-conf
+Source5:        make_webpack.sh
+
+Patch0:         000-update-npm-packages.patch
+Patch1:         001-router-basename.patch
+Patch2:         002-update-default-config.patch
+Patch3:         003-RPM-spec-and-webpack.patch
+
+BuildArch:      noarch
+BuildRequires:  nodejs
+
+Requires:	httpd-filesystem
+Requires:	nginx-filesystem
+Suggests:	httpd
+
+
+%description
+Vector is an open source on-host performance monitoring framework which exposes
+hand picked high resolution system and application metrics to every engineer’s
+browser. Having the right metrics available on-demand and at a high resolution
+is key to understand how a system behaves and correctly troubleshoot
+performance issues.
+
+
+%prep
+%setup -q -T -D -b 0 -n vector-%{vector_version}
+%setup -q -T -D -b 1 -n vector-%{vector_version}
+%setup -q -T -D -b 2 -n vector-%{vector_version}
+
+%patch0 -p1
+%patch1 -p1
+%patch2 -p1
+%patch3 -p1
+
+
+%build
+true
+
+
+%install
+install -d %{buildroot}%{_datadir}/%{name}
+cp -aT dist %{buildroot}%{_datadir}/%{name}
+
+# webserver configurations
+install -D -p -m 0644 %{_sourcedir}/vector-httpd-conf %{buildroot}%{_sysconfdir}/httpd/conf.d/%{name}.conf
+install -D -p -m 0644 %{_sourcedir}/vector-nginx-conf %{buildroot}%{_sysconfdir}/nginx/default.d/%{name}.conf
+
+
+%check
+node_modules/jest/bin/jest.js
+
+
+%files
+%dir %{_datadir}/%{name}
+%{_datadir}/%{name}
+
+%config(noreplace) %{_sysconfdir}/httpd/conf.d/%{name}.conf
+%config(noreplace) %{_sysconfdir}/nginx/default.d/%{name}.conf
+
+%license LICENSE
+%doc CHANGELOG.md README.md
+
+
+%changelog
+* Wed May  8 2019 Andreas Gerstmayr <agerstmayr@redhat.com> 2.0.0-0.1.beta.1
+- initial Vector package
